bin/cron.php

```diff
@@ -0,0 +1,93 @@
+<?php
+
+/**
+ * Add this file to you system cron list to run every minute
+ * eg. `* * * * * * php /var/www/bin/cron.php`
+ */
+
+(PHP_SAPI !== 'cli' || isset($_SERVER['HTTP_USER_AGENT'])) && die('Cron Job: comment out the first line in the cron.php file to be able to run from browser');
+
+define('BASE_PATH', realpath(__DIR__ . '/../'));
+define('CRON_PATH', realpath(BASE_PATH . '/customs/cron'));
+define('LOG_PATH', realpath(BASE_PATH . '/storage'));
+
+// Composer Autoloader
+$loader = require BASE_PATH . '/vendor/autoload.php';
+require_once BASE_PATH . '/api/config.php';
+require CRON_PATH . '/CronJob.php';
+require __DIR__ . '/log.php';
+
+use Directus\Bootstrap;
+use Directus\Collection;
+use Directus\Util\StringUtils;
+use Directus\Database\TableSchema;
+use Directus\Database\TableGatewayFactory;
+use Directus\Database\TableGateway\BaseTableGateway;
+
+$container = new \Slim\Helper\Set();
+$container->singleton('zendDb', function () {
+    return Bootstrap::get('ZendDb');
+});
+$container->singleton('log', function () {
+    return new Log('cron');
+});
+$container->singleton('hookEmitter', function () {
+    return Bootstrap::get('hookEmitter');
+});
+
+TableSchema::setConnectionInstance($container->get('zendDb'));
+TableSchema::setConfig(Bootstrap::get('config'));
+BaseTableGateway::setHookEmitter($container->get('hookEmitter'));
+TableGatewayFactory::setContainer($container);
+
+$container->set('schemaManager', Bootstrap::get('schemaManager'));
+
+$log = $container->get('log');
+
+if (!function_exists('should_execute')) {
+    eval('$minuteOfDay = ' . date('z*1440+G*60+i;'));
```

Comment:
Any reason why is using `eval`?

Reply:
:) That is some pretty interesting piece of code :)
First I use php date format to get the `z` (Day of the year), `G` (24 hour) and `i` (current minute) of the current date, but then I have to do some calculations on the day and hour so it is relative to minutes.

So in the end when the result is assigned to `$minuteOfDay` it is the current minute relative to the amount of minutes since the beginning of this year.

It can work without eval:
`
$G = date('G')*60;
$z = date('z')*1440;
$minuteOfDay = date(i) + $z + $G;
`

Reply:
`date('z') * 1440 + date('G') * 60 + ((int) date('i'))` because `date('i')` returns minutes with leading zeros.